### PR TITLE
Fixing button of type=reset so it trigger via enter/space

### PR DESCRIPTION
**Pull request by LFDanLu**
Closes https://github.com/adobe/react-spectrum/issues/2825
## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

Go to the http://localhost:9003/?path=/story/form--form-with-reset story and make changes to the form. Try reseting the form via pressing Enter/Space when focused on the reset button. It should reset the form values 

## 🧢 Your Project:

RSP

### CODE REVIEW

File: packages/@react-aria/interactions/src/usePress.ts [hunk 1]
```diff
@@ -770,7 +770,7 @@ function shouldPreventDefault(target: Element) {
 }
 
 function shouldPreventDefaultKeyboard(target: Element) {
-  return !((target.tagName === 'INPUT' || target.tagName === 'BUTTON') && (target as HTMLButtonElement | HTMLInputElement).type === 'submit');
+  return !((target.tagName === 'INPUT' || target.tagName === 'BUTTON') && ((target as HTMLButtonElement | HTMLInputElement).type === 'submit' || (target as HTMLButtonElement | HTMLInputElement).type === 'reset'));
```
Comment (LFDanLu) [inline packages/@react-aria/interactions/src/usePress.ts:773]: What do we think about this change? Do we want to add `onReset` to Form as well to mirror `onSubmit`? Or do we prefer that the user do the following approach for their reset buttons: https://github.com/adobe/react-spectrum/blob/dfe72ee076f7ca27728d0b6b4c1ef86e11191ece/packages/%40react-spectrum/form/stories/Form.stories.tsx#L575

Comment (snowystinger) [reply packages/@react-aria/interactions/src/usePress.ts:773]: Ok, so there are some issues with form reset in React, have you taken a look at my old PR that addressed submit? it used to handle reset as well, and if I recall correctly, the radio group and a few other controlled components just couldn't handle reset

Comment (LFDanLu) [reply packages/@react-aria/interactions/src/usePress.ts:773]: I found this following thread: https://github.com/adobe/react-spectrum/pull/904#discussion_r467188676. Did some digging online and I see mention of controlled components having problems with reset and for the most part this is handled via a onReset callback. I can punt on this PR until we add onReset or I can implement it here, but IMO this at least fixes the uncontrolled case and doesn't break any additional functionality

Comment (snowystinger) [reply packages/@react-aria/interactions/src/usePress.ts:773]: yeah, it's fine to add it, it looks like we also need to add a Type to Forms for onReset?

I think the hesitation for adding it was that even people who are using our components uncontrolled would run into issues. For instance, NumberField and RadioGroup are technically controlled in our implementation. People can choose to use our components in a controlled or uncontrolled way, but under the hood, WE are always using them in a controlled way. And during a reset, no event is fired, so our components don't know to update.

Comment (LFDanLu) [reply packages/@react-aria/interactions/src/usePress.ts:773]: makes sense, I'll see about putting this all into an issue. Def can document the above in the Forms docs

File: packages/@react-spectrum/form/docs/Form.mdx [hunk 2]
```diff
@@ -79,8 +79,8 @@ technology as a form landmark region to which assistive technology can navigate.
 
 ## Events
 Form accepts an `onSubmit` and `onReset` prop for handling form submission and resets respectively. In cases where your form elements' values are controlled, you must handle
-resetting each element's value to their original default via the callback provided via `onReset`. This is particularly important when using form elements such as RadioGroup and NumberField
-whose values are inherently controlled through their implementation.
+resetting each element's value to their original default via the callback provided via `onReset`. This is particularly important when using form elements such as Radio, RadioGroup,
```
Comment (snowystinger) [inline packages/@react-spectrum/form/docs/Form.mdx:82]: I mean, this is going to matter for way more components than it'll just work for. Probably better not to be explicit about any of them and just say it's important.

Other fields I can think of are, ColorField/Area/Wheel, CheckboxGroup, Sliders, Combobox, SearchField, SearchWithin, Picker, not sure if any are problematic or if there are others I'm forgetting

Comment (LFDanLu) [reply packages/@react-spectrum/form/docs/Form.mdx]: Sounds good, updated to remove the list of components

**Comment by LFDanLu** [comment]
Will need some more work/thought for handling our inherently controlled components, closing for now

**Pull request closed without merging**